### PR TITLE
Handled week & day interval in population of mrr events

### PR DESCRIPTION
**Pull request by allouis**
refs https://github.com/TryGhost/Team/issues/635

This is to ensure we don't break migrations for any sites which have
imported external subscriptions which have an interval of 'week' or
'day

### CODE REVIEW

File: core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ module.exports = createTransactionalMigration(
...
             }
+
+            if (interval === 'week') {
+                return amount * 4;
+            }
+
+            if (interval === 'day') {
+                return amount * 30;
+            }
         }
```
Comment (daniellockyer) [inline core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js:40]: @allouis So we've seen this migration fail before where the subscription interval was `week` and obviously this PR fixes that, but would it be useful to have some sort of error/ignoring for when it's still not handled? I realise we're probably handling all cases now but..... 😅

File: core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js [hunk 2]
```diff
@@ -29,6 +29,14 @@ module.exports = createTransactionalMigration(
             if (interval === 'month') {
                 return amount;
             }
+
+            if (interval === 'week') {
+                return amount * 4;
+            }
+
+            if (interval === 'day') {
+                return amount * 30;
```
Comment (daniellockyer) [inline core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js:38]: Are the `30` and `4` just above too general to "the average week or month" or doesn't it really matter for us?

File: core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js (hunk 1 above)
Comment (allouis) [reply core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js:40]: So the only valid values from stripe are: "day, week, month or year." - but I think you're right, we should probably error if we encounter an unknown one

File: core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js (hunk 2 above)
Comment (allouis) [reply core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js:38]: I spoke with Matt and he's happy with these - "Yep, they’re edge cases anyway"

File: core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js (hunk 1 above)
Comment (daniellockyer) [reply core/server/data/migrations/versions/4.0/25-populate-members-paid-subscription-events-table.js:40]: I foresee `decade` or something 😆

**Review by daniellockyer — APPROVED** [review]

**Pull request closed without merging**